falcon/_typing.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2021-2023 by Vytautas Liuolia.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Private type aliases used internally by Falcon.."""
+
+from __future__ import annotations
+
+from enum import auto
+from enum import Enum
+import http
+from http.cookiejar import Cookie
+import sys
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Literal,
+    Mapping,
+    Optional,
+    Pattern,
+    Protocol,
+    Tuple,
+    TYPE_CHECKING,
+    TypeVar,
+    Union,
+)
+
+# NOTE(vytas): Mypy still struggles to handle a conditional import in the EAFP
+#   fashion, so we branch on Py version instead (which it does understand).
+if sys.version_info >= (3, 11):
+    from wsgiref.types import StartResponse as StartResponse
+    from wsgiref.types import WSGIEnvironment as WSGIEnvironment
+else:
```

Comment:
The `else:` branch looks like a genuine coverage miss. Should we add a pragma, or do we need to update `.coveragerc` to omit `falcon/_typing.py` instead of `falcon/typing.py`?

Reply:
don't we collect coverage for <3.11?

Reply:
No, we don't.

I think you simply need to update the "omit" section in `.coveragerc` to omit `_typing.py`, not `typing.py`.

Reply:
well we can add a ignore comment in the else. will do

Reply:
If that's all that's needed for the whole file, then remove the file from coveragerc omissions.

Reply:
oh I think the issue was that the protocol are not liked by coverage. that's why it was added

Reply:
Oh yeah, that was another reason!

Reply:
I have update `.coveragerc` in the web UI. Let's see how it goes now.